src/main/java/seedu/address/model/achievement/AchievementRecord.java

```diff
@@ -61,37 +177,52 @@ public void setXp(Xp xp) {
     public void resetData(AchievementRecord newData) {
         requireNonNull(newData);
 
-        setXp(newData.getXp());
-        setLevel(newData.getLevel());
+        displayOption = newData.displayOption;
+        xp = newData.xp;
+        level = newData.level;
+        numTaskCompleted = newData.numTaskCompleted;
+        nextDayBreakPoint = newData.nextDayBreakPoint;
+        numTaskCompletedByDay = newData.numTaskCompletedByDay;
+        xpValueByDay = newData.xpValueByDay;
+        nextWeekBreakPoint = newData.nextWeekBreakPoint;
+        numTaskCompletedByWeek = newData.numTaskCompletedByWeek;
+        xpValueByWeek = newData.xpValueByWeek;
     }
 
     /**
-     * Updates the Xp field of this {@code AchievementRecord} with new xp value.
+     * Updates all fields of this {@code AchievementRecord} with new xp value.
      * Triggers the update of Level with xp.
      */
-    public void updateXp(Integer xp) {
-        requireNonNull(xp);
+    public void updateAchievementsWithNewXp(Integer newXp) {
```

Comment:
done

Reply:
about testing the achievements being reset, it can be tested with Junit tests (which i'll write for 1.4). but if another team want to test without writing code, they'll need to change the system time of their computer.